go/libraries/doltcore/sqle/tables.go

```diff
@@ -128,9 +134,34 @@ func (t *DoltTable) WithIndexLookup(lookup sql.IndexLookup) sql.Table {
 	}
 }
 
+// doltTable returns the underlying doltTable from the current session
+func (t *DoltTable) doltTable(ctx *sql.Context) (*doltdb.Table, error) {
+	root := t.lockedToRoot
+	if root == nil {
```

Comment:
I don't love this bit of state to support `AS OF` queries. I think this would be better as a separate type that has an immutable root value

Reply:
Yeah, I'll look for ways to get rid of this as I keep digging